cmd/test/flag.go

```diff
@@ -0,0 +1,55 @@
+package test
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/giantswarm/microerror"
+	"github.com/spf13/cobra"
+
+	"github.com/giantswarm/standup/pkg/gsclient"
+)
+
+const (
+	flagKubeconfig = "kubeconfig"
+	flagEmail      = "email"
+	flagEndpoint   = "endpoint"
+	flagInCluster  = "in-cluster"
+	flagPassword   = "password"
+	flagProvider   = "provider"
+	flagRelease    = "release"
+)
+
+type flag struct {
+	Kubeconfig string
+	Email      string
+	Endpoint   string
+	InCluster  bool
+	Password   string
+	Provider   string
+	Release    string
+}
+
+func (f *flag) Init(cmd *cobra.Command) {
+	cmd.Flags().StringVarP(&f.Kubeconfig, flagKubeconfig, "k", "", `The path to the kubeconfig for the control plane.`)
+	cmd.Flags().StringVarP(&f.Email, flagEmail, "e", "", `The email of the user account used to authenticate with the GS API.`)
+	cmd.Flags().StringVarP(&f.Endpoint, flagEndpoint, "n", "", `The endpoint of the target control plane's API.`)
+	cmd.Flags().BoolVarP(&f.InCluster, flagInCluster, "i", false, `True if this program is running in a Kubernetes cluster and should communicate with the API via the injected service account token.`)
+	cmd.Flags().StringVarP(&f.Password, flagPassword, "w", "", `The password of the user account used to authenticate with the GS API.`)
+	cmd.Flags().StringVarP(&f.Provider, flagProvider, "p", "", fmt.Sprintf(`The provider of the target release. Possible values: <%s>`, strings.Join(gsclient.AllProviders(), "|")))
+	cmd.Flags().StringVarP(&f.Release, flagRelease, "r", "", fmt.Sprintf(`The semantic version of the release to be tested.`))
+}
+
+func (f *flag) Validate() error {
+	if f.Kubeconfig == "" && !f.InCluster || f.Kubeconfig != "" && f.InCluster {
```

Comment:
The flags for connecting to the CP Kubernetes API. This allows us to test with a kubeconfig or run it in the cluster and connect to the parent API.